### DISCUSSION TITLE
Speed limit panel new implementation

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#452] [Q&A] [answered]
- The panel with `!` at top right (image 1) appears all the time displaying the current speed limit.
The panel option is in navigation display settings (image 2).

- The warning at top center appears when go above the speed limit by the selected threshold.
The threshold option is in navigation advanced settings (image 3).

<img src="https://github.com/devemux86/cruiser/assets/3484020/d4b870cd-4c7e-46db-80cc-b40e93fc93e8" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/b91ce71d-b2f2-477f-84eb-5c8cd89fc934" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/406f57ef-533b-4df3-80cd-dfa9513c3c53" width="33%"/>

**Comment by devemux86** [comment]
> The panel with ! at top right appears all the time

Which is - depending on how much other panels are displayed in which order and what the current light conditions are - very hard to recognize. Therefore I did not activate it before.

> The warning at top center

I would LOVE to have this displayed "all the time" instead of the small panel on the right.
It could then start to flash, if one is above the set limit...

Is this possible???

_Originally posted by @bios4 in https://github.com/devemux86/cruiser/discussions/441#discussioncomment-8543513._

**Comment by devemux86** [comment]
@bios4 we could think about a different display of speed limits.

Perhaps using different colors in the top-center circle?
- Display: black circle
- Warning: red circle

**Reply by bios4** [reply]
> Display: black circle

No, please NO - I would urge to display it the VERY SAME way as is looks like in real life (and on the recent screenshots), means with the red border and the black value!

I am sure it would look weird with a different color to what is known and what road users all over the world (at least Europe 😁) are used to...

> Warning: red circle

When one is above the limit threshold, you could start to flash it with inverted colors - white border/value on red ground.

**Reply by devemux86** [reply]
@bios4 we need to somehow differentiate the display from the warning.

Animated texts and flashing colors are not suitable for quick glances while driving or for battery and performance.

**Reply by bios4** [reply]
What about my addition with inverted colors ^^ (but then without the flashing)?

<img alt="" src="https://github.com/devemux86/cruiser/assets/145460147/2e6d873e-6260-436e-9074-ac7fad92d600" width="33%">
<img alt="" src="https://github.com/devemux86/cruiser/assets/145460147/9f7377df-d132-4473-aaba-210490a12edb" width="33%">

Quick & dirty suggestions with Paint.NET ^^ 😁

**Comment by WolfgangZ42** [comment]
Just for more ideas here my current solution with the overlayed app "Tempo Master":
<img width="500" src="https://github.com/devemux86/cruiser/assets/121806238/7be86790-9ea6-4b08-accf-41196e3ac1fd"/>

**Reply by bios4** [reply]
> Just for more ideas here my current solution with the overlayed app "Tempo Master"

👍 - looks almost identical to what I would love to have in Cruiser (w/o external/additional app)...

How is the speed limit displayed when you are driving above the limit?
Does it change its look to make you aware of it?

**Reply by devemux86** [reply]
@WolfgangZ42 thanks we need more ideas. :+1: 

Your image displays the current speed limit (in red circle).
What happens when you are above this, is there a visual warning?

**Reply by WolfgangZ42** [reply]
No - it doesn't change the display. But it gives a "beep" every 10 seconds you are over the limit. It's adjustable in various ways.

**Comment by umpfmpf** [comment]
I have now made a small lap: I have been waiting for the speed display for a long time. Now I have noticed that it takes a long time for the allowed speed to be displayed correctly. Most of the time I am out of a zone with a speed limit until the display changes. There is usually a delay of 5-30 seconds.

**Reply by devemux86** [reply]
@umpfmpf do you mean the speed limit info panel at top-right?

Please provide the route so we can examine the OpenStreetMap tags in the route.
This is the OpenStreetMap tag: https://wiki.openstreetmap.org/wiki/Key:maxspeed

Also please mention how you calculated the route?
Only some route services include max speed in the routes.

**Reply by umpfmpf** [reply]
Yes, the speed limit on top right.
I calculated the route offline with a router. 
I Start at 50.187646,9.194782
And end at
50.092242,9.20753

**Reply by devemux86** [reply]
@umpfmpf what router, BRouter (offline)?

Please export and attach a gpx navigation or json or kurviger file.
It is much easier to check the route. :slightly_smiling_face:

**Reply by umpfmpf** [reply]
Yes, with Brouter offline.
[test.gpx.txt](https://github.com/devemux86/cruiser/files/14363794/test.gpx.txt)

**Reply by devemux86** [reply]
@umpfmpf is this [route](https://brouter.de/brouter-web/#map=13/50.1432/9.2088/standard&lonlats=9.194859,50.187643;9.206714,50.093322&profile=car-fast)?
Where exactly you did not see max speed?

Checking the above BRouter link and the "Data" tab, there are several roads without defined max speed.
(some at start and more near the end)

<img src="https://github.com/devemux86/cruiser/assets/3484020/ce2b5470-99b1-4f4e-bde2-a3ba380087d9" width="40%"/>

**Reply by devemux86** [reply]
I see with this route example that there is area for improvements in Cruiser reader of BRouter data.
So there will be improvements in the next update.

**Reply by MotoUKRider** [reply]
I have observed (in my capacity as a novice OSM editor) that there are sections of road I *know* to have a speed-limit change yet the segment "features" are not accurate to where I know the sign-post to be.  There are also gaps, so the source data doesn't appear to be perfect.  This would go some way towards explaining a delay with speed-limit display updates - I have found more than 1 issue with the OSM data that could easily be blamed on the front-end app :clown_face: 

Testing a simulated route with a few local speed variations, I find that Cruiser's panel updates when it should but that there are big gaps in the source data - and roads seem to be broken into many segments, some missing a "speed" value, others are out of date.  For example, coming into a village near me - there's a B-Road that's a "National" (60mph) speed-limit, but the segment extends about 300yrds past the residential 30mph sign that's been there a few years.

Even a year ago when I tested some routes using GraphHopper, there were gaps and incorrect values - confirming my theory that the app can only display the data it has to work with (if any!).

**Reply by devemux86** [reply]
Example: we have road segments A -> B -> C -> D:
- **A -> B**: 50 km/h in OSM data
- **B -> C**: _no max speed in OSM data_
- **C -> D**: 50 km/h in OSM data

Question: how an app should use the B -> C segment, the one without max speed in OSM data.
Should it display no max speed to user or continue to show the last max speed of A -> B segment?

**Reply by bios4** [reply]
> Should it display no max speed to user or continue to show the last max speed of A -> B segment?

You should definitely NOT(!!!) display the "last known max. speed of A -> B" segment!
What if the section where the OSM data is missing has a lower limit?

If the data you rely on does not contain the required information, better display nothing than a wrong value...
My 0,05.

**Reply by devemux86** [reply]
> You should definitely NOT(!!!) display the "last known max. speed of A -> B" segment!
What if the section where the OSM data is missing has a lower limit?

> If the data you rely on does not contain the required information, better display nothing than a wrong value...

That's what Cruiser is doing now. :slightly_smiling_face:

**Reply by bios4** [reply]
> That's what Cruiser is doing now.

...which is perfectly fine from my POV! 👍

**Reply by MotoUKRider** [reply]
I agree and would have replied the same: I know of several locations nearby that are 50 or 60mph roads but drop to a 40mph for barely a mile because there is a small cluster of homes along the route.  There's also a missed 30-60mph change at another location - it's not perfect data, but at least the current method is fairly easy to update and reflected in near real-time.  I would say that a blank display is best until valid data is available.

For now, we'll have to rely on those round metal panels stuck on metal sticks by the side of the road :smile:

**Reply by 0709wiwiwi** [reply]
If you keep the max speed in my region based only on the completely unreliable data fitted in open street maps Belgium might one day manage to reduce its immense national debt,  The only thing that still can be done here very fast  is the constant changing of the speed limits. Most osm contributors  don't so update all this silly changes anymore.  Welcome, my country is sure counting on your contributions too. Soon even the wheelchair users will be probably be moving too fast.

**Comment by umpfmpf** [comment]
In your list I see that the speeds are correct. I noticed that on the street too. But with a strong delay.  That means that I am out of a 50kmh zone with a length of 500m until 50kmh is displayed. Then goes on again 500m. Everything recognized, but only too late. Up to 1 time, there were almost 1km of offset in a piece of forest.

**Reply by devemux86** [reply]
@umpfmpf there are some more tags that I need to add in Cruiser reader of BRouter data, coming in next update.

**Comment by DolomitesTom** [comment]
nothing to comment... thank god i'm not interested in speed limits... I don't see them at all :-)... congenital red-white blindness

**Comment by 0709wiwiwi** [comment]
Although I have immense reservations about the (local) reliability of the osm max speed data, there is still the problem of what to show when a street is divided into several osm edited sections whereby the speed indications are too often missing.
The preference here was to show nothing or thus as (! --) max speed. A speed indication is so missing very often. 
Alternative? 
Road sections with a known speed limit are shown by the well-known road code signal red border ring & black numbers. 
If a next osm road section(s) is missing this data, continue to show the last shown sign with a black border ring & faded.

**Comment by umpfmpf** [comment]
I have now simulated the route and then started  driving it. In the simulation, exact is shown the same as on the street. However, the speed displayed is still very different from reality.

**Reply by devemux86** [reply]
@umpfmpf what do you mean "reality"?

Max speed in apps depends on what exists in OpenStreetMap database.

Please post a [BRouter website](https://brouter.de/brouter-web/) link and we can see immediately what maxspeed tags exist on each road segment.

**Reply by umpfmpf** [reply]
I mean the real driving on the road with the actual speedlimits. 
Here a testing route 
[Test (9,7 km).gpx.txt](https://github.com/devemux86/cruiser/files/14389160/Test.9.7.km.gpx.txt)

**Reply by devemux86** [reply]
@umpfmpf is [this route](https://brouter.de/brouter-web/#map=13/50.1573/9.2136/standard&lonlats=9.195299,50.187384;9.199204,50.183896;9.199696,50.182946;9.199703,50.180527;9.201214,50.176686;9.205158,50.17283;9.206852,50.168388;9.206655,50.167325;9.203225,50.164993;9.203576,50.163976;9.205514,50.161983;9.205678,50.159449;9.204634,50.158109;9.204629,50.156292;9.206205,50.155216;9.207866,50.152975;9.212509,50.151893;9.214532,50.150935;9.214803,50.14908;9.212766,50.14726;9.214757,50.143685;9.219633,50.14136;9.221973,50.139234;9.220808,50.135633;9.218354,50.13176;9.218653,50.131308;9.218354,50.13176;9.217053,50.130491;9.209961,50.127904;9.200434,50.128275;9.197651,50.127939;9.197441,50.126555;9.198062,50.125014;9.199502,50.124131;9.199126,50.123052&profile=car-fast)?

What is the problem exactly? Looking at its data table it has maxspeed except near start and end:

<img src="https://github.com/devemux86/cruiser/assets/3484020/58a45922-109a-4604-a7fa-b576bcc0ed0a" width="50%"/>

**Reply by umpfmpf** [reply]
I have drawn what the comparison is like: the left line is real, the right line is cruiser
<img src="https://github.com/devemux86/cruiser/assets/105640513/e31daee0-1796-46a2-b009-c912f3ec09fb" width="50%"/>

**Reply by devemux86** [reply]
@umpfmpf have you compared with [BRouter link](https://brouter.de/brouter-web/#map=13/50.1573/9.2136/standard&lonlats=9.195299,50.187384;9.199204,50.183896;9.199696,50.182946;9.199703,50.180527;9.201214,50.176686;9.205158,50.17283;9.206852,50.168388;9.206655,50.167325;9.203225,50.164993;9.203576,50.163976;9.205514,50.161983;9.205678,50.159449;9.204634,50.158109;9.204629,50.156292;9.206205,50.155216;9.207866,50.152975;9.212509,50.151893;9.214532,50.150935;9.214803,50.14908;9.212766,50.14726;9.214757,50.143685;9.219633,50.14136;9.221973,50.139234;9.220808,50.135633;9.218354,50.13176;9.218653,50.131308;9.218354,50.13176;9.217053,50.130491;9.209961,50.127904;9.200434,50.128275;9.197651,50.127939;9.197441,50.126555;9.198062,50.125014;9.199502,50.124131;9.199126,50.123052&profile=car-fast) and its Data table?

Is there something wrong or different from that table in Cruiser?

**Reply by umpfmpf** [reply]
Now I've compared the speeds in Brouter online. **There** is the error. 
The display in Cruiser is exactly what Brouter transmits. 
However, this differs from the actual permitted maximum speed when driving. 
If I run Google Maps next to it, there values and the change points between the speeds are correct.

**Reply by devemux86** [reply]
> The display in Cruiser is exactly what Brouter transmits.
However, this differs from the actual permitted maximum speed when driving.

@umpfmpf thanks very much for the check! :+1: 
It was important to understand where exactly is the difference.

Routers use the speed limits that are stored in OpenStreetMap database. And the apps get the same data.
So if there is a difference somewhere it's usually because is stored wrong and need to be fixed by all users?

**Reply by umpfmpf** [reply]
Where and how can I fix ?

**Reply by devemux86** [reply]
@umpfmpf please check the [OpenStreetMap Beginners' guide](https://wiki.openstreetmap.org/wiki/Beginners%27_guide):

_"If you just want to edit as soon as possible, visit https://www.openstreetmap.org/, zoom to an area, and click "edit". (You'll need an account first). The [default editor](https://wiki.openstreetmap.org/wiki/ID) has a built-in guide that will explain the basics. You can also click the built-in "help" button at any time."_

---

For example here is an OpenStreetMap link for one of the road sections of your route, you can see all its data:
https://www.openstreetmap.org/way/173344561

**Reply by 0709wiwiwi** [reply]
As I had established myself, the osm data is so erroneous that this max speed feature is virtually useless.  Sorry for the technical work and implementation done but the osm data is worthless.

**Reply by devemux86** [reply]
@0709wiwiwi everyone uses the same data (also the overlay apps).

OpenStreetMap can only improve if we all invest some time and improve our known areas.
This is always true for all free open products. :slightly_smiling_face:

**Reply by WolfgangZ42** [reply]
> everyone uses the same data (also the overlay apps)

That's true Emux! 

Of course, I tried out the speedlimit display and soon noticed the discrepancy between TempoMaster and Cruiser. During the short drive, I had the impression that the display was only offset by one section of the route. Whenever TempoMaster displays a new speed limit, Cruiser adopts the previous one; sometimes, of course, both are the same. Wrong stacking? 
Unfortunately, I hadn't recorded the test, so I ran a Lockito simulation of the @umpfmpf track. Check it out, it's just like my first impression. A bit boring 8 minutes - but you can fast forward: https://youtu.be/czw6SF4XtHQ

**Reply by Ciutnik** [reply]
Doesn't Mapillary automatically set speed limits based on road signs recognized in photos? I read a long time ago that these are street lamps, hydrants, traffic lights...

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the detailed test!

We may have some index wrong and must fix it, that's why I ask for specific details so can understand better.
I will check it. :slightly_smiling_face:

**Reply by 0709wiwiwi** [reply]
> OpenStreetMap can only improve if we all invest some time and improve our known areas.

Sorry but at least I'm not going to keep tweaking this further. The changes are so frequent locally here that it is almost impossible to keep up. The only important thing for the government is to generate a lot of revenue. For instance, our government boasts that by hiring extra staff, they will soon be able to issue 9 million fines a year. Kapo behaviour ? Must be because these types are apparently reproducing faster than what the much higher government ever envisaged with the order "go and multiply".

**Reply by WolfgangZ42** [reply]
> @Ciutnik Doesn't Mapillary automatically set speed limits based on road signs

I asked Copilot (~ChatGPT):
Does Mapillary read traffic signs?

Copilot:
Mapillary does not automatically read traffic signs during recording. However, when you capture street views using Mapillary, those images are uploaded to the database and can be viewed by other users. If you encounter incorrectly recognized traffic signs, you can manually report them using the following steps:

Select an Image: Go to the image view in the Mapillary app.
Show Options: Click the three dots in the lower right corner of the image.
Report Image: Choose the “Report Image” option.
While direct editing of traffic signs within Mapillary is not currently supported, you can use the image reporting feature to flag any inaccuracies.

If you’re interested in editing traffic signs for OpenStreetMap, you can freely use the images and data from Mapillary. It provides integrations with the RapiD Editor, iD Editor, and JOSM to view image content and AI-extracted point data such as traffic signs, road features, and infrastructure. 🚦📸

**Reply by Ciutnik** [reply]
Thank you @WolfgangZ42 for the investigation carried out 🙂 
But I was hoping that clever algorithms would do it automatically. Maybe someone will one day hire Artificial Intelligence for this job. He is perfect for this job 👍

**Reply by MotoUKRider** [reply]
I've made a few changes to roads I know have incorrect (or missing) data. I started by setting the actual speed-limit value, then got creative by breaking the nodes and setting the short bits where a reduction for houses was in effect.  It's good that the data can be edited without too much issue, and we only have to wait a day or so before we can pull-down the "tile" from BRouter to enjoy the new data.  I doubt I'll make a career out of, though!

**Reply by devemux86** [reply]
> we only have to wait a day or so before we can pull-down the "tile" from BRouter to enjoy the new data

One of the many advantages of BRouter. :+1:

**Reply by devemux86** [reply]
After a detailed check of what BRouter has in Data table against the route response,
indeed there is an offset in the route details, like the max speed!

We'll see with @afischerdev if the response needs improvement.

But for now I will fix it in the Cruiser reader in the next update.

---

@umpfmpf thanks for the report and everyone for the tests! :+1:

**Reply by devemux86** [reply]
@umpfmpf this should be fixed in the latest:
- https://github.com/devemux86/cruiser/discussions/451

Now the read data is synchronized with that in the BRouter Data table.

**Reply by devemux86** [reply]
@umpfmpf I forgot to mention that you need to _**recalculate**_ your routes,
so that there are processed again correctly by the new improved reader.

**Reply by umpfmpf** [reply]
I have now corrected the speed limits in openstreetmap. 
Now I'm curious to see what it looks like in a few days. 
At the moment it is very exactly what is stored in Brouter when I drive off the route. 
And thanks for the fact that the speed limit display also works in the simulation, which saves a lot of fuel when testing.

**Reply by umpfmpf** [reply]
It took 2 days, but now the new speed limits have been adopted. 
Now the display in Cruiser also matches the signage on the street. 
Should mean: if everyone finds the error improving it, we will have a really good navigation at some point with the correct speed limit display. 
So it is a bit of the task of the users to maintain them...
@devemux86: very good work

**Reply by devemux86** [reply]
@umpfmpf thanks again for the report and the tests! :+1:

**Reply by 0709wiwiwi** [reply]
> Should mean: if everyone finds the error improving it, we will have a really good navigation at some point with the correct speed limit display.

I would not trust the business.

**Reply by umpfmpf** [reply]
@0709wiwiwi 
I drove a little lap yesterday. 
I found that most streets are actually relatively precise. 
Only on b- c- or even smaller streets does the display become imprecise. 
And you can save them as favorites during driving and improve them later if you want to.
For me it is mainly relevant on the main streets where traps can be found.

**Reply by devemux86** [reply]
Let's not forget that we almost lost OpenAndroMaps, the story is here:
- https://www.openandromaps.org/en/updates-suspended

Fortunately Christian and the other members of OpenAndroMaps continue their excellent work:
- https://www.openandromaps.org/en/new-maps-new-features

**Comment by MotoUKRider** [comment]
The availability of speed-limit data on roads is varied to say the least.  As part of my efforts to "port" my map theme into BaseCamp, I have added the speed-limit to the mouse-hover (tool-tip) which very easily lets me see where road segments are missing speed data.

*Spoiler-alert* : It's quite a lot, at least in the UK.
As I wrote earlier in this thread, if the source data isn't there, the app can only help you so much.  It's just good to know that it can be fixed with a little learning and some time - rather than be locked into data (or themes) that can't be altered no matter how bad/incorrect/out-of-date :smile_cat: 

![basecamp-speedlimit](https://github.com/devemux86/cruiser/assets/125696109/36494e14-93cf-495f-856c-3df65f820b67)

**Reply by MotoUKRider** [reply]
It doesn't matter (to the average user) *where* the data comes from, so long as it's reliable and fairly up-to-date.  I believe CoPilot (when I subscribed to it) used a 3rd-party dataset (there were lots of copyright credits on the "About" screen) and updated every few months.

If BRouter is using the OSM "maxspeed" as a source, then there will always be gaps - and I've found that the gaps are literally that: A segment within a long road is missing a value, or a particular segment passes a residential area and the lower speed-limit has not been updated.  That's nobody's fault and everybody's fault :laughing:  Cruiser could, unfairly, be blamed for being unreliable in that regard if a user is in an area that is not of interest to editors of OSM.

The data needs to be there, if only for route duration calculation - you can't rely on road-types as always having specific speed-limits but I accept there will be a need to average-out the calculations because there is no value to work with.  It's never going to be perfect, but I'm at least wiser as to the "hows" and "whys" with my under-the-hood investigations!

**Reply by WolfgangZ42** [reply]
There is a URL to control the OSM settings of maxspeed: [](http://mijndev.openstreetmap.nl/~peewee32/maxspeed/Maxspeed.htm) - don't know if it works for Northern Ireland ... in Germany its helpful to identify gaps:
<img width="500" src="https://github.com/devemux86/cruiser/assets/121806238/52b3e2c8-6fdd-4282-ac6f-af2b4f9532d5"/>
a try for you @MotoUKRider:

<img width="500" src="https://github.com/devemux86/cruiser/assets/121806238/599eb5b0-5382-4111-8fe3-714b5ef93622"/>

**Reply by MotoUKRider** [reply]
@WolfgangZ42 Thanks for that example - that website is quite useful!  Obviously it's in km but I was able to "nose around".

@0709wiwiwi Lower speed limits are certainly welcome in built-up areas but most of these silly limits are a result of a) residents campaigning and/or b) incident statistics.  In an ideal world, we'd drive according to the conditions of the road within the (well sign-posted) limits and be able to stop in the distance we can see to be clear on our side of the road :police_officer:   For everything else, Denali make good after-market horns :trumpet: 

You were only speeding a little bit?  That's OK, the 5yr-old you hit is only a little bit dead.

**Comment by devemux86** [comment]
New panel to display the speed limit:

- Speed limit is displayed at the same position in different colors for:
  - Display
  - Warning (when exceeded)

- Current speed also changes color when the speed limit is exceeded

- Available in "Settings | Navigation | Panels | Speed limit" as before

- Voice announcement when speed limit exceeded works as before
"Settings | Voice guidance | Announcement | Speed limit"

- Old speed limit panel on the right was removed for more free space

Display|Warning
-|-
![Screenshot_1732711219](https://github.com/user-attachments/assets/e71d3c07-1d4f-4e25-acde-03ac52e908dd)|![Screenshot_1732874156](https://github.com/user-attachments/assets/da05dc29-fc87-40e5-a235-9b0073cb7614)

**Reply by devemux86** [reply]
The new panel to display the speed limit is available in **[Cruiser 5.1.0-b11](https://github.com/devemux86/cruiser/discussions/671)**.

**Comment by DolomitesTom** [comment]
> Speed limit is displayed at the same position in different colors for:
    Display
    Warning (when exceeded)
Available in "Settings | Navigation | Panels | Speed limit" as before

where do I now find the threshold where I set that a warning should only appear when I am 40 km/h above?

**Reply by devemux86** [reply]
It is still in "Settings | Navigation | Advanced | Speed limit tolerance".

**Comment by devemux86** [comment]
Another improvement is that current speed also changes color when the speed limit is exceeded.

I updated the screenshots [above](https://github.com/devemux86/cruiser/discussions/452#discussioncomment-11394482).

**Comment by MotoUKRider** [comment]
Should I be able to see this on a simulated route calculated with internal BRouter?
I've tested numerous places (with known speed values in OSM) but don't see the display in either orientation.

**Reply by devemux86** [reply]
@MotoUKRider Speed limit will also be displayed in simulation in the next update.

**Comment by devemux86** [comment]
- New panel to display the speed limit is available in https://github.com/devemux86/cruiser/discussions/684

**Comment by MotoUKRider** [comment]
Just noticed that the Speed-limit roundel gets auto-enabled when updating to the latest version from the Play store.

This happened on my Samsung S20 when I went out for a ride the other day - and just decided to update the backup S9 device: Checked the setting first, it was "off".  After updating: "on".  Odd!

**Reply by devemux86** [reply]
@MotoUKRider There was a needed change for the driving profiles.
We can disable it again if we do not need it.

**Reply by MotoUKRider** [reply]
@devemux86 Yes, it caught me by surprise as the first couple of roads don't have a limit, so it appeared automagically as I reached Warp Factor 7 on the first "big road".

**Comment by devemux86** [comment]
> is this also working in simulation mode or only while navigating?

@DolomitesTom Speed limit also appears in simulation. But it must exist in OSM data and the route.

> Can this tempolimit only be show when speed + limit threshold is exceeded? At least it doesn't show anything for me in simulation mode. I would like to test this "Blitzdings" (flash thing).

Current speed limit implementation that was agreed with the users can be seen in this long discussion.

You participated in this discussion [above](https://github.com/devemux86/cruiser/discussions/452#discussioncomment-8556411).
Should we change the speed limit implementation again?

If you are not interested in the speed limit panel, you can turn it off.
Displaying the panel only when you exceed the speed limit, that is the benefit?
That's why the voice speed limit warning exists, you can enable it in voice settings.

**Reply by DolomitesTom** [reply]
ok thanks, normally I don't need it, it is just because I drive through the Swiss soon, then maybe I will care for it, otherwise it gets expensive.

> Displaying the panel only when you exceed the speed limit, that is the benefit?

For me yes.

Voice warning sometimes can be overheard due to bluetooth or volume level issues.

**Reply by devemux86** [reply]
@DolomitesTom If you don't use the speed limit voice warning and have the speed limit panel off,
then how you will know the speed limit in your area?

Looking randomly the screen to see if the speed limit panel appears when you exceed the speed?

Just look your actual speed panel, when speed is over the speed limit, then it's painted in red:
(the 62 km/h in the screenshot)

<img src="https://github.com/user-attachments/assets/aee71b6d-3980-4f93-97c5-6ff637610031" width="50%"/>

**Comment by DolomitesTom** [comment]
> If you don't use the speed limit voice warning and have the speed limit panel off,
then how you will know the speed limit in your area?

I simply don't care because I am mostly far beyond anyway. I would hear the "Voice Pointing finger" permanently and would then become blunt.



> Looking randomly the screen to see if the speed limit panel appears when you exceed the speed?

I am looking quite often peripherally at the screen to have a quick prediction for the curve. And then the white and red prohibition sign just gets on my nerves.


> Just look your actual speed panel, when speed is over the speed limit, then it's painted in red:
(the 62 km/h in the screenshot)

Maybe this works in Swiss temporarily. Is this also turning into red when the speed panel is turned off?

**Reply by devemux86** [reply]
@DolomitesTom 

> Is this also turning into red when the speed panel is turned off?

Speed panel must be visible to display the speed or become red when exceed the speed limit.

**Reply by DolomitesTom** [reply]
> Speed panel must be visible to display the speed or become red when exceed the speed limit.

That's not too comfortable to my point of view. You don't need to change something just for me, just to understand: why this limitation?

**Reply by devemux86** [reply]
@DolomitesTom All the panels at the right are information panels.
We enable them when we want to see constantly their information.

**Reply by devemux86** [reply]
Some of the information panels may turn red when there is a warning in their data.
e.g. current speed, battery level

**Reply by MotoUKRider** [reply]
The "sudden appearance" of an icon (like the speed-limit warning) may be, ahem, more "helpful" than the icon just changing colour.

I recall that CoPilotGPS had a 2-switch config for "Display" and "Warning" so it would only appear if you were enjoying the throttle a bit too much.

**Reply by devemux86** [reply]
@MotoUKRider For warning panels this makes sense.
It also described [below](https://github.com/devemux86/cruiser/discussions/452#discussioncomment-13365779) as a possible improvement.

**Comment by devemux86** [comment]
Other [apps](https://osmand.net/docs/user/widgets/info-widgets/#speedometer) have options "Always" and "When exceeded", for the warning panel.
So we can offer something similar too.

**Reply by DolomitesTom** [reply]
you mean for the panel to appear? yes please, the option "when exceeded" is great...

**Reply by devemux86** [reply]
For the warning panel at the center of the screen.

**Comment by devemux86** [comment]
New speed limit panel options will be available in the next Cruiser:

<img src="https://github.com/user-attachments/assets/ee15bd7d-5ce3-4b3a-ae6c-6d0674ebddc8" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
- Speed limit panel new implementation is available in https://github.com/devemux86/cruiser/discussions/820
